### PR TITLE
[WIP] Update wxGTK and use GTK3 version

### PR DESCRIPTION
**Pull request by abbradar**
###### Motivation for this change

Part of our slow move to GTK3 where available.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I cannot build this right now because `nix` keeps failing with `Too many heap sections: Increase MAXHINCR or MAX_HEAP_SECTS` and I'm not sure what to do with it. I've built this some time before on staging though.

Can't test properly either because `glibc` is updated on master and locale databases are incompatible.

### CODE REVIEW

**Comment by abbradar** [comment]
Rebased, ran several applications and confirmed they work.

This will require frequent rebases so I'd like to have this merged faster rather than later; however this still requires more testing; I'll try to reserve some time for this but probably won't be able to cover it fully.

File: pkgs/applications/graphics/displaycal/default.nix [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
- }:
-buildPythonPackage {
-  name = "displaycal-3.5.0.0";
+{ buildPythonPackage, stdenv, fetchurl, pkgconfig
+, libXext, libXxf86vm, libX11, libXrandr, libXinerama, libXScrnSaver
+, argyllcms, wxPython, numpy
+}:
+
+buildPythonPackage rec {
+  name = "DisplayCAL-${version}";
```
Comment (FRidh) [inline pkgs/applications/graphics/displaycal/default.nix]: The `name` attribute is added by `buildPython*` and should therefore be removed. Just add `pname` and `version`.

**Comment by FRidh** [comment]
You should not need to fix all packages that depend on the GTK change.

File: pkgs/applications/graphics/displaycal/default.nix (hunk 1 above)
Comment (abbradar) [reply pkgs/applications/graphics/displaycal/default.nix]: I keep forgetting that one, thanks!

**Comment by pSub** [comment]
What remains to be done here? Would be nice to have the new version of wxGTK.

**Comment by pSub** [comment]
Would be nice to get this merged. The wxGtk update blocks at least the update of filezilla since several releases.

**Comment by FRidh** [comment]
@abbradar I suggest to get this in as soon as possible, considering the branch off in 4 weeks.

**Comment by jokogr** [comment]
@abbradar are you still working on this?

**Comment by abbradar** [comment]
This became sooo stale while I wasn't here; it's a pity but it needs to be redone almost from scratch and I don't have needed time now.

**Pull request closed without merging**